File: DemoAppObj-C/YahooSearchKitDemo/View Controllers/YSKHeaderFooterViewController.m
//
//  YSKHeaderFooterViewController.m
//  YahooSearchKitDemo
//
//  Copyright 2015 Yahoo! Inc.
//  Licensed under the terms of the zLib license. See LICENSE file at the root of this project for license terms.
//

#import "YSKHeaderFooterViewController.h"
#import "YSKCustomHeader.h"
#import "YSKCustomFooter.h"
#import "YSKLandingPageViewController.h"
#import <YahooSearchKit/YahooSearchKit.h>

@interface YSKHeaderFooterViewController () <YSLSearchViewControllerDelegate>

@property (nonatomic, weak) IBOutlet UIButton *whiteThemeButton;
@property (nonatomic, weak) IBOutlet UIButton *darkThemeButton;
@property (nonatomic, weak) IBOutlet UIButton *transparentThemeButton;

@property (nonatomic,strong) UIColor *borderColor;

@end

@implementation YSKHeaderFooterViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.borderColor = [UIColor colorWithRed:207/255.0f green:207/255.0f blue:207/255.0f alpha:1.0f];
}

#pragma mark - UI Action Handlers

- (IBAction)whiteThemeButtonTapped:(id)sender {
    
    YSLSearchViewControllerSettings *settings = [YSLSearchViewControllerSettings new];
    YSLSearchViewController *searchViewController = [[YSLSearchViewController alloc] initWithSettings:settings];
    searchViewController.delegate = self;
    YSKCustomHeader *headerView = [[YSKCustomHeader alloc] initWithTheme:YSCHeaderViewThemeWhite];
    searchViewController.headerView = headerView;
    YSKCustomFooter *footerView = [[YSKCustomFooter alloc] initWithTheme:YSCFooterViewThemeWhite];
    searchViewController.footerView = footerView;
    UIApplication.sharedApplication.statusBarStyle = UIStatusBarStyleDefault;
    searchViewController.modalTransitionStyle = UIModalTransitionStyleCrossDissolve;
    [self presentViewController:searchViewController animated:YES completion:nil];
}

- (IBAction)darkThemeButtonTapped:(id)sender {
    
    YSLSearchViewControllerSettings *settings = [YSLSearchViewControllerSettings new];
    YSLSearchViewController *searchViewController = [[YSLSearchViewController alloc] initWithSettings:settings];
    searchViewController.delegate = self;
    YSKCustomHeader *headerView = [[YSKCustomHeader alloc] initWithTheme:YSCHeaderViewThemeDark];
    searchViewController.headerView = headerView;
    YSKCustomFooter *footerView = [[YSKCustomFooter alloc] initWithTheme:YSCFooterViewThemeDark];
    searchViewController.footerView = footerView;
    searchViewController.modalTransitionStyle = UIModalTransitionStyleCrossDissolve;
    [self presentViewController:searchViewController animated:YES completion:nil];
}

- (IBAction)transparentThemeTapped:(id)sender {
    
    YSLSearchViewControllerSettings *settings = [YSLSearchViewControllerSettings new];
    settings.enableTransparency = YES;
    YSLSearchViewController *searchViewController = [[YSLSearchViewController alloc] initWithSettings:settings];
    
    // Screenshot
    /*
     UIGraphicsBeginImageContextWithOptions(view.bounds.size, false, UIScreen.mainScreen().scale)
     self.view.drawViewHierarchyInRect(self.view.frame, afterScreenUpdates: true)
     var snapshot:UIImage! = UIGraphicsGetImageFromCurrentImageContext()
     UIGraphicsEndImageContext()
     */
    
    UIImage *snapshot = [UIImage imageNamed:@"BGimage.png"];
    
    // Setting it to landing page
    YSKLandingPageViewController *landingPage = [YSKLandingPageViewController new];
    landingPage.backgroundImage = snapshot;
    searchViewController.landingPageViewController = landingPage;
    searchViewController.hideLandingPage = NO;
    
    searchViewController.delegate = self;
    YSKCustomHeader *headerView = [[YSKCustomHeader alloc] initWithTheme:YSCHeaderViewThemeTransparent];
    searchViewController.headerView = headerView;
    YSKCustomFooter *footerView = [[YSKCustomFooter alloc] initWithTheme:YSCFooterViewThemeTransparent];
    searchViewController.footerView = footerView;
    searchViewController.modalTransitionStyle = UIModalTransitionStyleCrossDissolve;
    [self presentViewController:searchViewController animated:YES completion:nil];
}

#pragma mark - YSLSearchViewControllerDelegate methods

- (void)searchViewControllerDidTapLeftButton:(YSLSearchViewController*)searchViewController {
    UIApplication.sharedApplication.statusBarStyle = UIStatusBarStyleLightContent;
    [searchViewController dismissViewControllerAnimated:YES completion:nil];
}

- (YSLQueryAction)searchViewController:(YSLSearchViewController*)searchViewController actionForQueryString:(NSString*)queryString {
    return YSLQueryActionDefault;
}

@end

File: DemoAppSwift/YahooSearchKitDemo/View Controllers/YSKHeaderFooterViewController.swift
//
//  YSKSearchBarViewController.swift
//  YahooSearchKitDemo
//
//  Copyright 2015 Yahoo! Inc.
//  Licensed under the terms of the zLib license. See LICENSE file at the root of this project for license terms.
//

import UIKit

class YSKHeaderFooterViewController : UIViewController, YSLSearchViewControllerDelegate {
    
    var selectedSearchOptions:YSKViewControllerSearchOptions!
    let borderColor = UIColor(red: 207/255, green: 207/255, blue: 207/255, alpha: 1.0).CGColor
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.selectedSearchOptions = YSKViewControllerSearchOptions.SearchOptionDefault
    }
    
    // MARK: UI Action Handlers
    @IBAction func whiteThemeButtonTapped(sender: UIButton) {
        let settings:YSLSearchViewControllerSettings = YSLSearchViewControllerSettings()
        
        let searchViewController = YSLSearchViewController(settings: settings)
        searchViewController.delegate = self
        let headerView: YSKCustomHeader! = YSKCustomHeader(theme: YSCHeaderViewTheme.White)
        searchViewController.headerView = headerView
        let footerView: YSKCustomFooter! = YSKCustomFooter(theme: YSCFooterViewTheme.White)
        searchViewController.footerView = footerView
        UIApplication.sharedApplication().statusBarStyle = .Default
        searchViewController.modalTransitionStyle = UIModalTransitionStyle.CrossDissolve
        presentViewController(searchViewController, animated: true, completion:nil)
    }
    
    @IBAction func darkThemeButtonTapped(sender: AnyObject) {
        let settings:YSLSearchViewControllerSettings = YSLSearchViewControllerSettings()
        
        let searchViewController = YSLSearchViewController(settings: settings)
        searchViewController.delegate = self
        let headerView: YSKCustomHeader! = YSKCustomHeader(theme: YSCHeaderViewTheme.Dark)
        searchViewController.headerView = headerView
        let footerView: YSKCustomFooter! = YSKCustomFooter(theme: YSCFooterViewTheme.Dark)
        searchViewController.footerView = footerView
        searchViewController.modalTransitionStyle = UIModalTransitionStyle.CrossDissolve
        presentViewController(searchViewController, animated: true, completion: nil)
    }
    
    @IBAction func transparentThemeTapped(sender: UIButton) {
        let settings:YSLSearchViewControllerSettings = YSLSearchViewControllerSettings()
        settings.enableTransparency = true
        let searchViewController = YSLSearchViewController(settings: settings)
        
        // Screenshot
        /*
        UIGraphicsBeginImageContextWithOptions(view.bounds.size, false, UIScreen.mainScreen().scale)
        self.view.drawViewHierarchyInRect(self.view.frame, afterScreenUpdates: true)
        var snapshot:UIImage! = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        */
        
        let snapshot: UIImage! = UIImage(named: "BGimage.png")
        
        
        // Setting it to landing page
        let landingPage: YSKLandingPageViewController! = YSKLandingPageViewController()
        landingPage.backgroundImage = snapshot
        searchViewController.landingPageViewController = landingPage
        searchViewController.hideLandingPage = false
        
        searchViewController.delegate = self
        let headerView: YSKCustomHeader! = YSKCustomHeader(theme: YSCHeaderViewTheme.Transparent)
        searchViewController.headerView = headerView
        let footerView: YSKCustomFooter! = YSKCustomFooter(theme: YSCFooterViewTheme.Transparent)
        searchViewController.footerView = footerView
        searchViewController.modalTransitionStyle = UIModalTransitionStyle.CrossDissolve
        presentViewController(searchViewController, animated: true, completion:nil)
    }
    
    // MARK: YSLSearchViewControllerDelegate methods
    func searchViewControllerDidTapLeftButton(searchViewController: YSLSearchViewController!) {
        UIApplication.sharedApplication().statusBarStyle = .LightContent
        searchViewController.dismissViewControllerAnimated(true, completion: nil)
    }
    func searchViewController(searchViewController: YSLSearchViewController!, actionForQueryString queryString: String!) -> YSLQueryAction {
        return YSLQueryAction.ShowSuggestions
    }
    
}

